lib/interactive_viewer.ml

```diff
@@ -104,6 +104,64 @@ let change_summary z_patches : ui Lwd.t =
   in
   W.string ~attr:Notty.A.(fg lightcyan) operation_count
 
+
+(** Side by side diff view implementation **)
+
+type view_mode = SideBySide | Normal
+
+let view_mode = Lwd.var SideBySide
+
+let toggle_view_mode () =
+  match Lwd.peek view_mode with
+  | Normal -> Lwd.set view_mode SideBySide
+  | SideBySide -> Lwd.set view_mode Normal
+
+let rec split_and_align_hunk hunks mine_acc their_acc =
```

Comment:
There's a bug in the line numbers and in the aligning:
```
   1   (executable                             1   (executable
   2    (public_name diffcessible)             2    (public_name diffcessible)
   3    (name main)                            3    (name main)
   4 -  (libraries diffcessible cmdliner))     4   
   5                                           5 +  (libraries diffcessible cmdliner patch))
```

The added line is also line 4 !

Reply:
It seems that you didn't push your recent work.

Reply:
Hi @Julow, I haven't pushed it yet sorry 😅. I had to quickly work on my programming language design and implementation assignment before I got back to working on this PR. I can push my current progress tonight and hopefully finish by this Wednesday 👍

Reply:
Ok, good luck with your assignment :)